loki/transformations/tests/test_loop_blocking.py

```diff
@@ -0,0 +1,538 @@
+        loops) == num_loops + 1, f"Total number of loops transformation is: {len(loops)} but expected {num_loops + 1}"
+    assert len(
+        routine.variable_map) == num_vars + LOKI_LOOP_SLIT_VAR_ADDITION, f"Total number of variables after loop splitting is: {len(routine.variable_map)} but expected {num_vars + LOKI_LOOP_SLIT_VAR_ADDITION}"
+
+    filepath = tmp_path / (f'{routine.name}_{frontend}.f90')
+    function = jit_compile(routine, filepath=filepath, objname=routine.name)
+
+    a = np.zeros(n, order='F')
+    b = np.zeros(n, order='F')
+    function(a, b, n)
+    assert np.all(b == 0.), "b array should not be modified."
+    a_ref = np.linspace(1, n, n)
+    assert np.array_equal(a, a_ref), "a should be equal to a_ref=(1, 2, ..., n)"
+
+    clean_test(filepath)
+
+
+@pytest.mark.parametrize('frontend', available_frontends())
+@pytest.mark.parametrize('block_size', [117])
+@pytest.mark.parametrize('n', [500])
+def test_2d_splitting(tmp_path, frontend, block_size, n):
+    fcode = """
+    subroutine test_2d_splitting(a, b, n)
+      integer, intent(in) :: n
+      real(kind=8), intent(inout) :: a(n)
+      real(kind=8), intent(inout) :: b(n,n)
+      real(kind=8) :: c(n)
+      integer :: i
+      !$loki driver-loop
+      do i=1,n
+        a(i) = i
+        c(1) = a(i)
+        b(:,i) = a(i)
+      end do
+    end subroutine test_2d_splitting
+        """
+    routine = Subroutine.from_source(fcode, frontend=frontend)
+    loops = FindNodes(ir.Loop).visit(routine.ir)
+    num_loops = len(loops)
+    num_vars = len(routine.variable_map)
+    with pragmas_attached(routine, Loop):
+        loops = find_driver_loops(routine,
+                                  targets=None)
+    splitting_vars, inner_loop, outer_loop = split_loop(routine, loops[0], block_size)
+    loops = FindNodes(ir.Loop).visit(routine.ir)
+
+    assert len(
+        loops) == num_loops + 1, f"Total number of loops transformation is: {len(loops)} but expected {num_loops + 1}"
+    assert len(
+        routine.variable_map) == num_vars + LOKI_LOOP_SLIT_VAR_ADDITION, f"Total number of variables after loop splitting is: {len(routine.variable_map)} but expected {num_vars + LOKI_LOOP_SLIT_VAR_ADDITION}"
+
+    filepath = tmp_path / (f'{routine.name}_{frontend}.f90')
+    function = jit_compile(routine, filepath=filepath, objname=routine.name)
+
+    a = np.zeros(n, order='F')
+    b = np.zeros((n,n), order='F')
+    function(a, b, n)
+    a_ref = np.linspace(1, n, n)
+    b_ref = np.tile(a_ref, (n,1))
+    assert np.array_equal(a, a_ref), "a should be equal to a_ref=(1, 2, ..., n)"
+    assert np.array_equal(b, b_ref), "b should equal b_ref"
+
+    clean_test(filepath)
+
+
+
+@pytest.mark.parametrize('frontend', available_frontends())
+@pytest.mark.parametrize('block_size', [117])
+@pytest.mark.parametrize('n', [500])
+def test_3d_splitting(tmp_path, frontend, block_size, n):
+    fcode = """
+    subroutine test_3d_splitting(a, b, c, n)
+      integer, intent(in) :: n
```

Comment:
```suggestion
      implicit none
      integer, intent(in) :: n
```

Won't mark this in the rest of the file but please add to all the code snippets